db/migrate/20231218170631_add_rate_to_percentage_amount_details.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+class AddRateToPercentageAmountDetails < ActiveRecord::Migration[7.0]
+  class Fee < ApplicationRecord
+    belongs_to :charge, -> { with_discarded }, optional: true
+  end
+
+  def up
+    percentage_fees = Fee.joins(:charge).merge(Charge.percentage).where.not(amount_details: nil)
+
+    percentage_fees.find_each do |fee|
+      fee.update!(
+        amount_details: fee.amount_details.except!('per_unit_amount').merge(
+          rate: BigDecimal(fee.charge.properties['rate'].to_s)
+        )
```

Comment:
Style/TrailingCommaInArguments: Put a comma after the last parameter of a multiline method call. (https://rubystyle.guide#no-trailing-params-comma)